extensions/entsoe/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+#
+# Copyright (c) 2020, RTE (http://www.rte-france.com)
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+set(EXT_NAME ext-entsoe)
+
+set(EXT_VERSION_MAJOR ${IIDM_VERSION_MAJOR})
+set(EXT_VERSION_MINOR ${IIDM_VERSION_MINOR})
+set(EXT_VERSION_PATCH ${IIDM_VERSION_PATCH})
+set(EXT_VERSION ${EXT_VERSION_MAJOR}.${EXT_VERSION_MINOR}.${EXT_VERSION_PATCH})
+set(EXT_SOVERSION ${EXT_VERSION_MAJOR})
+
+set(EXT_SOURCES
+    src/EntsoeArea.cpp
+    src/EntsoeAreaXmlSerializer.cpp
+    src/EntsoeGeographicalCode.cpp
+    src/MergedXnode.cpp
+    src/MergedXnodeXmlSerializer.cpp
+    src/Xnode.cpp
+    src/XnodeXmlSerializer.cpp
+)
+
+set(UNIT_TEST_SOURCES
+    test/entsoe.cpp
+    test/EntsoeAreaTest.cpp
+    test/MergedXnodeTest.cpp
+    test/XnodeTest.cpp
+)
+
+# Shared library
+add_library(${EXT_NAME} SHARED ${EXT_SOURCES})
+set_target_properties(${EXT_NAME} PROPERTIES SOVERSION ${EXT_SOVERSION} VERSION ${EXT_VERSION} OUTPUT_NAME iidm-${EXT_NAME})
+target_include_directories(${EXT_NAME}
+    INTERFACE
+        $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+)
+target_link_libraries(${EXT_NAME} iidm)
+
+# Static library
+add_library(${EXT_NAME}-static STATIC ${EXT_SOURCES})
+set_target_properties(${EXT_NAME}-static PROPERTIES OUTPUT_NAME iidm-${EXT_NAME})
+target_include_directories(${EXT_NAME}-static
+    INTERFACE
+        $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+)
+target_link_libraries(${EXT_NAME}-static iidm)
+
+# Unit tests
+add_executable(unit-tests-${EXT_NAME} ${UNIT_TEST_SOURCES})
+target_link_libraries(unit-tests-${EXT_NAME} PRIVATE ${EXT_NAME} iidm-tests ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY})
+
+add_test(NAME ${EXT_NAME} COMMAND unit-tests-${EXT_NAME} ${BOOST_ARGS_SEPARATOR} --resources=${CMAKE_CURRENT_SOURCE_DIR}/resources)
+
+# Installation
+install(TARGETS ${EXT_NAME} ${EXT_NAME}-static
+    EXPORT iidm-targets
+    LIBRARY DESTINATION ${INSTALL_LIB_DIR}
+    ARCHIVE DESTINATION ${INSTALL_LIB_DIR}
```

Comment:
DESTINATION ${INSTALL_LIB_DIR} may be sufficient, not sure what is the difference of giving LIBRARY and ARCHIVE

Reply:
LIBRARY is used for dynamic library, ARCHIVE for static one